src/Symfony/Component/ArgumentResolver/ValueResolver/ValueResolverInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ArgumentResolver\ValueResolver;
+
+use Symfony\Component\ArgumentResolver\Exception\InvalidRawValueException;
+use Symfony\Component\ArgumentResolver\ArgumentMetadata\ArgumentMetadata;
+
+/**
+ * Responsible for resolving the value of an argument based on its metadata.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Robin Chalas <robin@baksla.sh>
```

Comment:
```suggestion
 * @author Robin Chalas <robin@baksla.sh>
 *
 * @template T of mixed
```
I wonder if it is worth to define the type of yielded data dynamically?